drv/ignition-server/src/main.rs

```diff
@@ -0,0 +1,243 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Server for interacting with Ignition Controllers.
+
+#![no_std]
+#![no_main]
+
+use drv_ignition_api::*;
+use drv_sidecar_mainboard_controller::ignition::*;
+use ringbuf::*;
+use userlib::*;
+
+task_slot!(FPGA, fpga);
+#[cfg(feature = "sequencer")]
+task_slot!(SEQUENCER, sequencer);
+
+#[allow(dead_code)]
+#[derive(Copy, Clone, Debug, PartialEq)]
+enum Trace {
+    None,
+    AwaitingMainboardControllerReady,
+    PortCount(u8),
+    PresenceUpdate(u64),
+}
+ringbuf!(Trace, 16, Trace::None);
+
+const TIMER_NOTIFICATION_MASK: u32 = 1 << 0;
+const TIMER_INTERVAL: u64 = 1000;
+
+#[export_name = "main"]
+fn main() -> ! {
+    let mut incoming = [0u8; idl::INCOMING_SIZE];
+    let mut server = ServerImpl {
+        controller: IgnitionController::new(FPGA.get_task_id()),
+        port_count: 0,
+        last_presence_summary: 0,
+    };
+
+    // This task is expected to run in an environment where a sequencer is
+    // responsible for configuring the mainboard controller/Ignition Controller
+    // logic. But for development purposes it may make sense for this assumption
+    // not to be true and run this task with something else guaranteeing that
+    // the mainboard controller (or something which looks like it) is present
+    // and ready.
+    cfg_if::cfg_if! {
+        if #[cfg(feature = "sequencer")] {
+            let sequencer =
+                drv_sidecar_seq_api::Sequencer::from(SEQUENCER.get_task_id());
+
+            // Poll the sequencer to determine if the mainboard controller is
+            // ready.
+            ringbuf_entry!(Trace::AwaitingMainboardControllerReady);
+            while !sequencer.mainboard_controller_ready().unwrap_or(false) {
+                hl::sleep_for(25);
+            }
+        }
+    }
```

Comment:
Using `cfg_if` is overkill here, since it's a single feature-based conditional:
```suggestion
    #[cfg(feature = "sequencer")]
    {
        let sequencer =
            drv_sidecar_seq_api::Sequencer::from(SEQUENCER.get_task_id());

        // Poll the sequencer to determine if the mainboard controller is
        // ready.
        ringbuf_entry!(Trace::AwaitingMainboardControllerReady);
        while !sequencer.mainboard_controller_ready().unwrap_or(false) {
            hl::sleep_for(25);
        }
    }
```

Reply:
I did not realize this syntax was an option. But thinking about how decorators work it makes sense, thanks!